### PR TITLE
prealloc linter

### PR DESCRIPTION
**Pull request by mythi**

### CODE REVIEW

File: cmd/bpf2go/main.go [hunk 1]
```diff
@@ -303,7 +303,7 @@ type target struct {
 }
 
 func printTargets(w io.Writer) {
-	var arches []string
+	arches := make([]string, len(targetByGoArch))
```
[the remark this answers is not in the document]
Comment (mythi) [reply cmd/bpf2go/main.go]: fixed

**Comment by lmb** [comment]
Is this really worth it? I can see the core change make sense, but the other one seems non-sensical to me. What problem does enabling this linter solve?

**Comment by ti-mo** [comment]
@lmb This is the same discussion, revived. It doesn't make things worse and allows catching potential optimizations in CI and during development if your editor runs golangci-lint automatically. Why is this not worth it?

**Comment by mythi** [comment]
> What problem does enabling this linter solve?

the slices that have known capacity can be preallocated. with this we never run into a case where the runtime would have to re-allocate a slice during a loop which is more optimal.

**Review by ti-mo — APPROVED** [review]
IMO this is good.

**Comment by lmb** [comment]
Sorry for being quiet for so long, last week was pretty busy with Linux Plumber's Conference taking up a lot of my time.

A linter is a rule that we always want to enforce, and so we turn it into code. In this case the rule is "always preallocate if you can". The reasons given are:

* It prevents possible performance problems
* It doesn't make things worse

No offense, but the second point isn't an argument at all, so I'll ignore it. To the first argument I'll reply: a possible performance problem is not really a thing. We can only tell what is a performance problem when we have real code, and that code is too slow, and we measure it. Quoting from the golangci-lint documentation:

```
  prealloc:
    # XXX: we don't recommend using this linter before doing performance profiling.
    # For most programs usage of prealloc will be a premature optimization.
```

So, without convincing reasons why we need this, enabling the linter is simply enforcing a rule for the rules sake. To me that makes it a dogma, which we should strive to avoid.

**Comment by mythi** [comment]
> Quoting from the golangci-lint documentation:

@lmb OK I had missed this part somehow. 

> enabling the linter is simply enforcing a rule for the rules sake

I think here comes the argument @ti-mo had: it does not make things worse and it may help automatically find potential issues.

But given the suggestion also from `golangci-lint` docs, I'm fine dropping this PR (or updating it to just improve the core part)

**Comment by lmb** [comment]
My preference would be to keep the core part but drop the linter.

**Pull request merged**